src/activity_handlers.cpp

```diff
@@ -689,9 +700,19 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
             liquid.charges = item::INFINITE_CHARGES;
             break;
         case LST_MAP_ITEM:
-            if( static_cast<size_t>( act.values.at( 1 ) ) >= source_stack.size() ) {
+            if( static_cast<size_t>( act.values.at( 1 ) ) >= source_stack.size() ){
                 throw std::runtime_error( "could not find source item on ground for liquid transfer" );
             }
+            break;
+        case LST_MONSTER:
+        Creature *c = g->critter_at(source_pos);
+        source_mon = dynamic_cast<monster *>(c);
+        if( source_mon == nullptr ){
```

Comment:
Could someone please help me with how the fill_liquid_do_turn function helps in activity_handlers.cpp?

Learned how to use breakpoints and looked at the c object and source_mon, both of them had the correct monster and location of it. But for some reason the throw at LST_MONSTER case is triggered even if source_mon isn't a null pointer.